src/pudl/workspace/setup.py

```diff
@@ -58,139 +46,38 @@ def get_defaults(
         os.environ["PUDL_OUTPUT"] = str(Path("~/pudl-work/output").expanduser())
         os.environ["PUDL_INPUT"] = str(Path("~/pudl-work/data").expanduser())
 
-    yaml_settings = _munge_legacy_yaml(
-        yaml_file=yaml_file, default_pudl_yaml=default_pudl_yaml
-    )
+    if input_dir:
+        os.environ["PUDL_INPUT"] = str(Path(input_dir).expanduser())
+    if output_dir:
+        os.environ["PUDL_OUTPUT"] = str(Path(output_dir).expanduser())
+
+    for env_var in ["PUDL_INPUT", "PUDL_OUTPUT"]:
+        if env_var not in os.environ:
+            raise RuntimeError(f"{env_var} environment variable must be set.")
 
-    # read from env vars
-    env_var_mapping = {
-        "pudl_in": os.getenv("PUDL_INPUT"),
-        "pudl_out": os.getenv("PUDL_OUTPUT"),
-    }
-    env_settings = {
-        key: str(Path(value))
-        for key, value in env_var_mapping.items()
-        if value is not None
-    }
-
-    # read from params
-    kwarg_mapping = {"pudl_in": input_dir, "pudl_out": output_dir}
-    kwarg_settings = {
-        key: str(Path(value))
-        for key, value in kwarg_mapping.items()
-        if value is not None
-    }
-
-    # Start with an empty settings, then override in order of precedence.
-    settings: dict[str, str] = {}
-    for settings_source in [yaml_settings, env_settings, kwarg_settings]:
-        settings |= settings_source
-
-    if not ("pudl_in" in settings and "pudl_out" in settings):
-        raise RuntimeError(
-            "Must set 'PUDL_OUTPUT'/'PUDL_INPUT' environment variables or provide valid yaml config file."
-        )
-
-    settings = derive_paths(settings["pudl_in"], settings["pudl_out"])
-
-    # override env vars so Dagster can see the most up-to-date configs
-    if output_dir or "PUDL_OUTPUT" not in os.environ:
-        os.environ["PUDL_OUTPUT"] = settings["pudl_out"]
-    if input_dir or "PUDL_INPUT" not in os.environ:
-        os.environ["PUDL_INPUT"] = settings["data_dir"]
+    settings = derive_paths(
```

Comment:
It sort of seems like we could inline `derive_paths` here too - then we could override other things beyond `input_dir` and `output_dir`.

I guess if everything is derived from the input/output dirs then that just allows us to accidentally override ourselves into weird states, though, so maybe this is better.

Reply:
I have inlined all of this and refactored somewhat. Generally I think the good idea is to reduce the number of tuning knobs so I think that having simply just input/output should be enough. There should be no need to have too much flexibility around this IMO.